cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/SecureStoreHandler.java

```diff
@@ -58,29 +58,31 @@ public SecureStoreHandler(SecureStore secureStore, SecureStoreManager secureStor
     this.secureStoreManager = secureStoreManager;
   }
 
-  @Path("/key")
+  @Path("/keys/{key-name}")
   @PUT
-  public void create(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace)
-    throws BadRequestException, IOException {
+  public void create(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace,
+                     @PathParam("key-name") String name) throws BadRequestException, IOException {
+    if (!EntityId.isValidStoreKey(name)) {
+      throw new BadRequestException("Improperly formatted name. The name can contain lower case alphabets, " +
+                                    "numbers, _, and -");
+    }
     SecureKeyCreateRequest secureKeyCreateRequest = parseBody(httpRequest, SecureKeyCreateRequest.class);
 
     if (secureKeyCreateRequest == null) {
-      throw new BadRequestException("Unable to parse the request.");
+      throw new BadRequestException("Unable to parse the request. The request body should be of the following format" +
+                                      "{\n" +
```

Comment:
this is great. However, any way we can make this future proof (if this structure ever changes in future)? Can we use a dummy object of a class?